src/services/restaurant.service.ts

```diff
@@ -1,79 +1,24 @@
+import { API_URL } from "constants/variables";
 import { ICard } from "constants/interfaces";
 
-export interface IFilterBy {
-  isNew?: boolean;
-  isPopular?: boolean;
-  isOpen?: boolean;
-}
-
-const query = async (filterBy: IFilterBy) => {
+const query = async (filterBy: string): Promise<ICard[]> => {
   try {
-    const response = await fetch("http://localhost:9090/api/restaurants");
+    const response = await fetch(`${API_URL}/restaurants?filterBy=${filterBy}`);
     if (!response.ok) {
       throw new Error(`Error loading restaurants. Status: ${response.status}`);
     }
```

Comment:
just bringing to your attention that in case somebody calls the query function without try catch block, the throwing of an error here could lead to a bad user experience